### PR TITLE
Refactor post-reportback & post-signup modals.

### PR DESCRIPTION
**Pull request by DFurnes**
### What does this PR do?

This pull request updates the post-reportback and post-signup modals to use the new modal implementation. These are the last two, so it also removes the old implementation! Nice! 🔥

### Any background context you want to provide?

One catch – moving the post-reportback modal into local component state makes the  "Show Reportback Affirmation" button in the admin bar difficult to implement. It also feels like that button makes less sense in a multi-uploader world, though, so it might be worth re-thinking more thoroughly in the future.

I kept the state for the post-signup modal in the Redux store (as `signups.showAffirmationModal`) since we trigger this modal as part of a thunked action, and it let's us more easily keep the "Show Affirmation Modal" button in the admin bar.

### What are the relevant tickets/cards?

N/A

### Checklist

* [ ] Documentation added for new features/changed endpoints.
* [ ] Added appropriate feature/unit tests.
* [ ] Added screenshot to PR description of related front-end updates on **small** screens.
* [ ] Added screenshot to PR description of related front-end updates on **medium** screens.
* [ ] Added screenshot to PR description of related front-end updates on **large** screens.

### CODE REVIEW

**Comment by weerd** [comment]
>  It also feels like that button makes less sense in a multi-uploader world, though, so it might be worth re-thinking more thoroughly in the future. 👍

**Comment by DFurnes** [comment]
Ah, [yep](https://styleci.readme.io/docs/change-log#section-14042018-april-fixer-updates)!

> Finally, the long deprecated "linting" config option has now been removed.

**Comment by DFurnes** [comment]
Ok, I removed that setting in 4f5caa74. Here's some further context from when it [was deprecated](https://styleci.readme.io/docs/change-log#section-27082016-linting-changes).

**Comment by DFurnes** [comment]
Welp, this turned into another crazy PR... but mostly deleted code!! 😁

screen shot 2018-04-16 at 12 59 43 pm

I'd like to move each of those standalone modal components in `components/Modal` into their own component directories, but didn't want to make this PR even wilder.

File: resources/assets/components/Modal/configurations/BlockModal.js [hunk 1]
```diff
@@ -1,40 +0,0 @@
...
-import ModalControls from '../ModalControls';
-import StaticBlock from '../../StaticBlock';
-import { parseContentfulType } from '../../../helpers';
-
-const BlockModal = props => {
-  const { json, closeModal } = props;
-
-  let children = <ContentfulEntry json={json} />;
-
-  if (parseContentfulType(json) === 'page') {
```
[the remark this answers is not in the document]
Comment (weerd) [reply resources/assets/components/Modal/configurations/BlockModal.js]: 🤷‍♂️

File: resources/assets/components/Modal/configurations/PostSignupModal.js [hunk 2]
```diff
@@ -2,9 +2,9 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import { CompetitionBlockContainer } from '../../CompetitionBlock';
 import SlideshowContainer from '../../Slideshow';
-import BlockModal from './BlockModal';
+import ContentfulEntry from '../../ContentfulEntry';
```
Comment (weerd) [inline resources/assets/components/Modal/configurations/PostSignupModal.js]: 👍

File: resources/assets/components/Modal/configurations/BlockModal.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (mendelB) [reply resources/assets/components/Modal/configurations/BlockModal.js]: I've seen this used in the link in [this block](https://phoenix-preview.dosomething.org/us/campaigns/give-spit-about-cancer#step-6yfXsr4iB2EW40u4yEquO4) for instance. I guess we can just add the `page` case to `ContentfulEntry` as you said?

[the remark this answers is not in the document]
Comment (DFurnes) [reply resources/assets/components/Modal/configurations/BlockModal.js]: Yeah, hmm. I think I'll add it into `ContentfulEntry` just to be safe.

**Comment by weerd** [comment]
So are some of those more specific modals, like the `VoterRegistrationModal` still needed because they're more standalone modals?

File: resources/assets/components/Modal/configurations/BlockModal.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (DFurnes) [reply resources/assets/components/Modal/configurations/BlockModal.js]: Ahhh yeah, good find @mendelB! 👍

**Comment by DFurnes** [comment]
> So are some of those more specific modals, like the `VoterRegistrationModal` still needed because they're more standalone modals?

Yeah, it's definitely subjective but that was the thought process! I left them since it seemed cleaner to reference [in the `ModalLauncher`](https://github.com/DoSomething/phoenix-next/blob/59c764e3c68480165b4a25c3df91a82116c2569e/resources/assets/components/Campaign/Campaign.js#L46) as a standalone component, and same with the post-signup modal.

In hindsight, I could also re-extract the `PostReportbackModal` into it's container so that we could easily reference it from the admin bar too, but I liked that inlining it fixes the problem of grabbing the right affirmation in a multi-uploader world.

File: resources/assets/components/Modal/configurations/BlockModal.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (DFurnes) [reply resources/assets/components/Modal/configurations/BlockModal.js]: Done in 8e47a420, and that modal now renders correctly on my local! ✅

**Comment by DFurnes** [comment]
Going to merge this once the build passes!

**Pull request merged**